### PR TITLE
Add the standardrb linter 

### PR DESCRIPTION
**Pull request by 0x1eef**
standardrb provides a relaxed set of linting rules that
provide the benefits of a linter without the headaches
that can come from the slightly excessive rubocop defaults.

New rules we'd like to adopt can be enabled in
`.rubocop.yml`, and similarly rules we don't want
to adopt can be disabled in `.rubocop.yml`.

Most rules were auto-correctable, a few were not 
and uncovered potential bugs. There are some disabled 
rules in `.rubocop.yml` - not all of them seem undesirable,
but perhaps something to fix in the future.

### CODE REVIEW

**Comment by 0x1eef** [comment]
@webmachine/ruby-maintainers 
What do you think about dropping <= 2.5 from the build matrix ? 2.6, and before that has already reached EOL. It would help bring this PR to a green build. If it is preferred to continue testing against those Ruby versions, I can looked into addressing the underlying issue.

File: lib/webmachine/constants.rb [hunk 1]
```diff
@@ -3,73 +3,73 @@
...
 
-  MATCHES_ALL = '*/*'.freeze
+  MATCHES_ALL = "*/*".freeze
 
-  GET_METHOD     = "GET"
-  HEAD_METHOD    = "HEAD"
-  POST_METHOD    = "POST"
-  PUT_METHOD     = "PUT"
-  DELETE_METHOD  = "DELETE"
+  GET_METHOD = "GET"
```
Comment (0x1eef) [inline lib/webmachine/constants.rb:31]: This might not be improvement. I can disable the indentation rules if that's preferred.

**Comment by bethesque** [comment]
Hi! Yes, I think 2.5 should be dropped. 2.7 is the lowest I'm using (which is embarrassingly old enough imo).

**Comment by bethesque** [comment]
With regards to the linting changes - I don't feel like I have an ownership level of this codebase that would permit me to approve that, but I think it's fine, and if it was my codebase, I'd say 👍🏽 It's probably @seancribbs call.

**Comment by seancribbs** [comment]
I didn't see anything objectionable in the style changes. My only question was whether the micro-optimization of single-quoted strings in the case where interpolation is unnecessary still applies. I'm guessing not?

**Comment by 0x1eef** [comment]
> My only question was whether the micro-optimization of single-quoted strings in the case where interpolation is unnecessary still applies. I'm guessing not?

I believe single quotes are the rubocop default, but not the standardrb default. I changed the rule to prefer single quotes instead: [commit](https://github.com/webmachine/webmachine-ruby/commit/df100211c76ff52a8f200cbd84626d289d80a77e). I can't speak to the performance issue, but I think the difference between the two is so small that the decision can be made based on other criteria.  

>  Yes, I think 2.5 should be dropped. 2.7 

Thanks ! Dropped [here](https://github.com/webmachine/webmachine-ruby/commit/0b5ea5d52f51d19093d215f6ca1ff629882fada4).

**Comment by orien** [comment]
We should consider adding a [`required_ruby_version`](https://guides.rubygems.org/specification-reference/#required_ruby_version) constraint to the gem spec. As these changes may intro incompatibilities with earlier versions of Ruby.

**Comment by bethesque** [comment]
Good idea.

**Review by orien — DISMISSED** [review]
Looks safe to me.

File: Guardfile [hunk 2]
```diff
@@ -1,11 +1,11 @@
 gemset = ENV['RVM_GEMSET'] || 'webmachine'
 gemset = "@#{gemset}" unless gemset.to_s == ''
 
-rvms = %W[ 1.8.7 1.9.2 1.9.3 jruby rbx ].map {|v| "#{v}#{gemset}" }
+rvms = %W[1.8.7 1.9.2 1.9.3 jruby rbx].map { |v| "#{v}#{gemset}" }
```
Comment (seancribbs) [inline Guardfile:4]: If we aren't going to support these old versions of Ruby in CI, we should remove them from this file. Does anyone use `guard` anymore when working with the project?

Comment (0x1eef) [reply Guardfile:4]: Usually, Guard is not part of my workflow.

Comment (bethesque) [reply Guardfile:4]: No

Comment (0x1eef) [reply Guardfile:4]: I'm not sure if we should just remove the `Guardfile` or not. It looks coupled to RVM, and the list of rubies being used are all end of life (except jruby). It seems easier to remove it than work with it - I don't have RVM to test this properly.

Comment (0x1eef) [reply Guardfile]: Alright - done in [this commit](https://github.com/webmachine/webmachine-ruby/commit/48125b18b1930935801a0df81c9eb075e5f076ec).

**Review by seancribbs — APPROVED** [review]
:shipit:

**Review by orien — APPROVED** [review]

**Pull request merged**